src/models/Order.js

```diff
@@ -78,13 +79,34 @@ const updateOrderStatus = async () => {
     // Get the current date and subtract __x   period of time
     const expiryTimeSinceCreation = new Date(Date.now() - timeDuration);
 
+    // Use aggregate to group orders by user before updating them
+    const ordersToUpdate = await Order.aggregate([
+      {
+        $match: {
+          orderStatus: 'pending',
+          createdAt: { $lte: expiryTimeSinceCreation },
+        },
+      },
+      { $group: { _id: '$user', orders: { $push: '$$ROOT' } } },
+    ]);
+    console.log('ordersToUpdate:', ordersToUpdate);
     // Update orders with a status of "pending" that were created x time ago or earlier,
     // and set the order status to "cancelled"
-    const result = await Order.updateMany(
-      { orderStatus: 'pending', createdAt: { $lte: expiryTimeSinceCreation } },
-      { $set: { orderStatus: 'cancelled' } }
-    );
-    console.log('Update result:', result);
+    for (const userOrders of ordersToUpdate) {
+      // Update orders for each user
+      await Order.updateMany(
+        { _id: { $in: userOrders.orders.map((order) => order._id) } },
+        { $set: { orderStatus: 'cancelled' } }
+      );
+
+      // Emit a message for the user
+      sendOrderCancelledEvent(io, {
+        userId: userOrders._id,
+        message: 'Your order has been cancelled.',
+      });
```

Comment:
`sendOrderCanceleldEvent` in `emitter.js` is only expecting 1 argument, the `userOrders` object of orders to emit info about.

```suggestion
      sendOrderCancelledEvent(userOrders);
```